cassiopeia/simulator/BirthDeathFitnessSimulator.py

```diff
@@ -0,0 +1,409 @@
+"""
+A general simulator for a birth-death tree simulation process, including 
+fitness. Takes any given distribution on waiting times and fitness 
+mutations.
+"""
+
+import networkx as nx
+import numpy as np
+from queue import PriorityQueue
+
+from typing import Callable, Dict, Optional, Union
+
+from cassiopeia.data.CassiopeiaTree import CassiopeiaTree
+from cassiopeia.simulator import TreeSimulator
+
+
+class BirthDeathFitnessError(Exception):
+    """An Exception class for all exceptions generated by the
+    GeneralBirthDeathSimulator
+    """
+
+    pass
```

Comment:
I think it would be better to have a `TreeSimulatorError` class that we can use for these exceptions